java/src/main/java/com/saucelabs/simplesauce/SauceOptions.java

```diff
@@ -1,134 +1,150 @@
-        return this;
-    }
-
-    public SauceOptions withSafari(final String version) {
-        String _version = version;
-        if (_version.isEmpty()) { _version = "latest"; }
-        browserName = BrowserType.SAFARI;
-        browserVersion = _version;
-        return this;
-    }
-
-    public SauceOptions withLinux() {
-        operatingSystem = "Linux";
-        return this;
-    }
-
-    public SauceOptions withWindows10() {
-        operatingSystem = "windows 10";
-        return this;
-    }
-    public SauceOptions withWindows8_1() {
-        operatingSystem = "Windows 8.1";
-        return this;
-    }
-    public SauceOptions withWindows8() {
-        operatingSystem = "Windows 8";
-        return this;
-    }
-
-    public SauceOptions withWindows7() {
-        operatingSystem = "Windows 7";
-        return this;
-    }
-
-    public SauceOptions withEdge() {
-        browserName = "Edge";
-        browserVersion = "18.17763";
-        return this;
-    }
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
 
-    //TODO notice the duplication below with edge.
-    //Maybe could be moved to a separate class so we can do withEdge().16_16299();
-    //Or withEdge().version(EdgeVersion.16_16299);
-    public SauceOptions withEdge18() {
-        withEdge();
-        browserVersion = "18.17763";
-        return this;
-    }
-    public SauceOptions withEdge17() {
-        withEdge();
-        browserVersion = "17.17134";
-        return this;
-    }
-    public SauceOptions withEdge16() {
-        withEdge();
-        browserVersion = "16.16299";
-        return this;
-    }
-
-    public SauceOptions withEdge15() {
-        withEdge();
-        browserVersion = "15.15063";
-        return this;
-    }
-
-    public SauceOptions withEdge14() {
-        withEdge();
-        browserVersion = "14.14393";
-        return this;
-    }
-
-    public SauceOptions withEdge13() {
-        withEdge();
-        browserVersion = "13.10586";
-        return this;
-    }
-
-    public SauceOptions withFirefox() {
-        browserName = "Firefox";
-        return this;
-    }
-
-    public SauceOptions withIE(IEVersion version) {
-        this.browserVersion = version.getVersion();
-        return this;
-    }
-
-    public SauceOptions withIE(String version) {
-        browserName = "IE";
-        browserVersion = version;
-        return this;
-    }
-
-    public SauceOptions withIE() {
-        browserName = "IE";
-        browserVersion = "latest";
-        return this;
-    }
+public class SauceOptions {
+    @Getter private MutableCapabilities seleniumCapabilities;
 
-    public SauceOptions withMac(MacVersion macVersion) {
-        operatingSystem = macVersion.label;
-        browserName = "Safari";
-        return this;
+    // w3c Settings
+    @Getter @Setter private String browserName = "chrome";
```

Comment:
I think Lombok has an option to make setters return `this`, which makes chaining easier. Consider using that?

Reply:
I don't really understand the appeal of the first over the second, and there are several additional complexities that come from relying on a fluent pattern. I get that it is a common pattern, I'm just reluctant to encourage it without a reason.
```java
sauceOptions.setBrowserName("chrome").
                         setBrowserVersion("77").
                         setPlatformName("Windows 10)
```
```java
sauceOptions.setBrowserName("chrome")
sauceOptions.setBrowserVersion("77")
sauceOptions.setPlatformName("Windows 10)
```

Reply:
@titusfortner I'm not recommending either of those approaches. The cleaner, simpler API approach is:
```
sauceOptions.withChrome("77").withWindows10();
```
One opportunity of messing up something here because obviously someone can pass in an unsupported Chrome version, but a risk we're willing to accept.

**Versus:**
```
sauceOptions.setBrowserName("chrome")
sauceOptions.setBrowserVersion("77")
sauceOptions.setPlatformName("Windows 10)
```
Three opportunities of messing something up here because each of the strings can be either fat fingered or incorrect.
And this is a simple example, adding even more Sauce Options continues to decrease the potential success of a user to get the `sauceOptions` object correct.